### PR TITLE
restore call-back for changed identities

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen 

Intended to supercede https://github.com/WhisperSystems/SignalServiceKit/pull/248

### CODE REVIEW

**Comment by michaelkirk** [comment]
rebased against latest  charlesmchen/identityManager

File: src/Messages/OWSIdentityManager.m [hunk 1]
```diff
@@ -302,47 +299,43 @@ - (BOOL)isTrustedIdentityKey:(NSData *)identityKey
 
 - (BOOL)isTrustedKey:(NSData *)identityKey forSendingToIdentity:(nullable OWSRecipientIdentity *)recipientIdentity
 {
-    OWSAssert(identityKey != nil);
+    OWSAssert(identityKey.length == 32);
```
Comment (charlesmchen) [inline src/Messages/OWSIdentityManager.m]: Let's constantize this.

File: src/Messages/TSMessagesManager.m [hunk 2]
```diff
@@ -202,7 +202,7 @@ - (NSString *)descriptionForDataMessage:(OWSSignalServiceProtosDataMessage *)dat
  */
 - (NSString *)descriptionForSyncMessage:(OWSSignalServiceProtosSyncMessage *)syncMessage
 {
-    NSMutableString *description = [[NSMutableString alloc] initWithString:@"CallMessage: "];
+    NSMutableString *description = [[NSMutableString alloc] initWithString:@""];
```
Comment (charlesmchen) [inline src/Messages/TSMessagesManager.m]: nit: `[NSMutableString new]` is shorter.

**Review by charlesmchen — APPROVED** [review]
LGTM

File: src/Messages/OWSIdentityManager.m (hunk 1 above)
Comment (michaelkirk) [reply src/Messages/OWSIdentityManager.m]: done!

File: src/Messages/TSMessagesManager.m (hunk 2 above)
Comment (michaelkirk) [reply src/Messages/TSMessagesManager.m]: done!

**Comment by michaelkirk** [comment]
In https://github.com/WhisperSystems/SignalServiceKit/commit/d475c58348a0a8c4ed9b0fcdbe352cd87e8ce269

**Pull request closed without merging**